builtin/common/misc_helpers.lua

```diff
@@ -159,6 +159,56 @@ function dump(o, indent, nested, level)
 end
 
 --------------------------------------------------------------------------------
+-- Allow indexing strings
+
+getmetatable"".__index = function(str, index)
+	local itype = type(index)
+	if itype == "string" then
+		return string[index] or str:find(index)
```

Comment:
l added a warning to lua_api because of this.
If one mod wants to break another one on purpose, there are several other ways, e.g. overriding minetest.register_node.